app/validators/csv_keyword_file_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CsvKeywordFileValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    @file = value
+
+    record.errors.add(attribute, I18n.t('activemodel.csv.errors.invalid_keyword_count')) unless valid_record_count?
+    record.errors.add(attribute, I18n.t('activemodel.csv.errors.invalid_file_type')) unless valid_content_type?
+  end
+
+  private
+
+  attr_reader :file
+
+  def valid_record_count?
+    CSV.read(file).count.between?(1, 1000)
```

Comment:
should we filter the `empty keyword` with `content.strip.empty?` 🤔

Reply:
Fixed in 0c4e8d7914b6fc71ddb701a3d3003cc1c1730124